src/gui_common/tooltip/microbe_editor/SelectionMenuToolTip.cs

```diff
@@ -68,6 +72,29 @@ public string DisplayName
         }
     }
 
+    /// <summary>
+    ///   Description of processes an organelle does if any.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     This supports custom format (for example: "Turns [glucose] into [atp]") where strings inside
```

Comment:
Would it make sense to namespace the bbcode, like: `[compound:glucose]` that way we could have a complex setup of  bbcode parsers (for example the thrivepedia might find this useful) attached to one rich text label?

Reply:
Want to mention that the format is not really bbcode (in the sense of tags) as the bracketed strings just serves as key so the regex could replace them with an actual bbcode string. That said, I think having a namespace is nice as it would allow different ways of parsing the same key which is very convenient in different contexts, though I feel adding such option could be reserved for later once features like Thrivepedia is implemented as for now the format usage is simple enough (stylizing compound name and inputs for tooltips).

Reply:
Even now this conflicts with normal bbcode, right? In that if there is a [b] in the text, your code would try to act as if that was an organelle name, right? I think that writing a simple proper parser would be a good idea even now to solve the code trying to work on unknown tags etc. Actually reading that Godot documentation you linked, I was kinda hoping that this would have been implemented as a bbcode extension to Godot, instead of just a transform on the actual text.

Reply:
Regarding conflict with bbcode tags it's possbile to make it so they will be ignored (returned unchanged by regex replace) since they don't match any compound names or inputs so they'll just work normally, I tried adding a strikethrough tag between one of the keys and it also work. Also it's unlikely a key and a bbcode tag has the same name, if it does I understand the need for a namespace. I'm assuming you're referring to [this](https://docs.godotengine.org/en/stable/tutorials/gui/bbcode_in_richtextlabel.html#custom-bbcode-tags-and-text-effects) part of the documentation, as far as I know it only allows adding custom tags for text effect purposes (which only operates per string characters, can't add images etc) which is why I decided to go with this approach.

Reply:
They work now because the code tries *any* bbtag to see if it is a valid compound name, and if it is, it is replaced. That's why the default tags still work.
That earlier idea about namespacing the tags would help avoid conflicts with inbuilt names *and* reduce the number of lookups "is this a compound name?" needed.
Yes that was the documentation I was referring to. So I guess this "preprocessor" approach to the bbcode is needed then.
Btw one more thing I noticed about the syntax, it's `[glucose]` which isn't valid bbcode, for consistency with other tags shouldn't we make the parser look for `[compound]glucose[/compound]` ? Or maybe even combining the namespaces: `[thrive:compound]glucose[/thrive:compound]` ?

Reply:
I feel like `[glucose]` is much simpler, but I'm fine with making it consistent with bbcodes so I'll try do that.

Reply:
Another idea would be to follow C# string formatting by using `{compound:glucose}` but I'm not sure if the automatic weblate C# template string checking would explode on that...
What I initially had in mind for the tooltips was that `{0}` etc. would be used in the template string and that would be replaced with the right compound names, but with your approach that probably isn't possible (my assumption was that the code doing the template text parsing knew what compound names etc. it would want to insert within the text placeholders).

Reply:
I'd guess probably yeah lol

> What I initially had in mind for the tooltips was that {0} etc. would be used in the template string and that would be replaced with the right compound names, but with your approach that probably isn't possible (my assumption was that the code doing the template text parsing knew what compound names etc. it would want to insert within the text placeholders).

That's also what I had in mind initially, but then thought that might get a bit messy as there's multiple places to maintain for ordering or something, with this it's much simpler and kind of like "all in one" I think.